scenes/globals/pause.gd

```diff
@@ -0,0 +1,114 @@
+extends Node
+
+enum System {
+	PLAYER_INPUT,
+	GAME
+	# For now is just these two but in the future we could have: Physics, Enemies, AI, etc
+}
+
+# Private vars
+var _pause_requests: Array[PauseRequest] = []
+
+# This should be an array of systems but it throws
+# a runtime error when using the filter function, so for now just "Array"
+var _paused_systems: Array = []
+
+
+# Public interface
+func pause_system(system: System, node: Node) -> void:
+	_add_request(system, node)
+
+
+func unpause_system(system: System, node: Node) -> void:
+	_clear_request(system, node)
+
+	_refresh_systems()
+
+
+func is_paused(system: System) -> bool:
+	return system in paused_systems()
```

Comment:
Why not use the `_paused_systems` Array directly?

Reply:
Because the _paused_systems array is an implementation detail. The contents of the array can actually be derived from the _pause_requests Array, so in a way, it's a cache. Having the method allows to change the implementation in the future if that's necessary.

That said, being the method is_paused defined inside the class, I guess it wouldn't hurt much to reference _paused_systems directly. I'm OK either way. Just say the word and I'll change it.